### ISSUE TITLE
long running queue triggered function does not dequeue

### ISSUE THREAD

**Issue by jchilde** [Azure/Azure-Functions#357]
Not sure if this is a cli issue or a function issue. I am trying to upload large files to sharepoint from azure storage.  

My function completes fine; however, it does not dequeue if it lasts a long time. I think the time is 5 minutes.  

I am NOT using a consumption based app. For example, a 20mb file takes 1 minute and dequeues.  A 1.2gb file takes 20mins and doesn't dequeue. 

Any ideas on how to troubleshoot since no errors.

**Comment by lindydonna** [comment]
CC @MikeStall

**Comment by DimaGershman** [comment]
I had similar issue with consumption plan - a queue-triggered function gives error after 5 minutes (`Microsoft.Azure.WebJobs.Host.FunctionTimeoutException`), but it kept firing a new function forever and didn't move the queue message to `poison`.
Eventually, I had to manually clear the queue to stop it and as a workaround, I added my own timeout within the function code.
It seems like a bug to me, isn't it?
